base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -265,17 +268,19 @@ where T: BlockchainBackend + 'static
                 header.version = constants.blockchain_version();
                 header.pow.target_difficulty = self.get_target_difficulty(*pow_algo).await?;
 
-                let transactions = async_mempool::retrieve(
-                    self.mempool.clone(),
-                    self.consensus_manager
-                        .consensus_constants()
-                        .get_max_block_transaction_weight(),
-                )
-                .await
-                .map_err(|e| CommsInterfaceError::MempoolError(e.to_string()))?
-                .iter()
-                .map(|tx| (**tx).clone())
-                .collect();
+                // We need to ensure that there is place to add an output, so that the miner can add a coinbase
+                let max_block_weight = self
+                    .consensus_manager
+                    .consensus_constants()
+                    .get_max_block_transaction_weight() -
+                    WEIGHT_PER_OUTPUT;
```

Comment:
A more maintainable approach would be to have `consensus_constants` define a `get_max_mineable_transaction_weight` function, since it's not inbound handler's job to work this out.

Reply:
I disagree with that. get_max_mineable_transaction_weight is not exactly a constant. Its defined as max_block_transaction_weight - weight_per_output. 

If we add that to the consensus code, we have consensus constants depended on fee, which goes the wrong way on the dependancy tree. 

While the input handler does not necessarily have the job to calculate the weight. It is this function's job to construct a block template. Calculating the block template I think includes calculating the max weight.

Reply:
Why not a method `.get_max_block_weight_excluding_coinbase()` which adds in this logic? Then it describes why we're removing these weights as well? You are only using constants in this formula, so I don't see where the fee gets involved.

Reply:
WEIGHT_PER_X
Currently lives as part of transactions::fee. 
The wallet needs access to the fee's. I dont think the wallet has access to the consensus crate. 

So we have two options here: 
Move the weights to consensus, but we have to let the wallet have consensus code as part of the wallet now. 

Keep the weights there, but then consensus has a dependancy on transactions::fee.

If one of those are fine, I will update it

Reply:
My feeling is that these should be in the consensus crate then. Two nodes will need to calculate the weight in the same manner, using the same WEIGHT_PER_X, otherwise they can't validate them. So then it feels like it's a consensus constant. Unless I'm mistaken?

Reply:
Agree, The weights can move to consensus constants. Changing weights would be a consensus change, so that makes sense

Reply:
I moved the weights to the consensus file. I did not move them into the network dependants as this would mean a dependancy on the consensus manager and the network type. I dont believe this would change between network types